src/log_surgeon/finite_automata/RegexAST.hpp

```diff
@@ -81,19 +82,23 @@ class RegexAST {
      */
     [[nodiscard]] virtual auto serialize() const -> std::u32string = 0;
 
-    [[nodiscard]] auto get_subtree_positive_tags() const -> std::set<uint32_t> const& {
+    [[nodiscard]] auto get_subtree_positive_tags() const -> std::vector<Tag const*> const& {
         return m_subtree_positive_tags;
     }
 
-    auto set_subtree_positive_tags(std::set<uint32_t> subtree_positive_tags) -> void {
+    auto set_subtree_positive_tags(std::vector<Tag const*> subtree_positive_tags) -> void {
         m_subtree_positive_tags = std::move(subtree_positive_tags);
     }
 
-    auto add_subtree_positive_tags(std::set<uint32_t> subtree_positive_tags) -> void {
-        m_subtree_positive_tags.merge(subtree_positive_tags);
+    auto add_subtree_positive_tags(std::vector<Tag const*> subtree_positive_tags) -> void {
```

Comment:
```suggestion
    auto add_subtree_positive_tags(std::vector<Tag const*> const& subtree_positive_tags) -> void {
```
There's no ownership transfer in this call so we should add const&

Reply:
Ah this makes sense. I was originally planning to use move semantics here, but it won't be useful for a vector of pointers. I.e., using `std::make_move_iterator`.